platform-services.md

```diff
@@ -0,0 +1,215 @@
+contains the code that queries the Android system location API
+([`android.location.LocationManager`](https://developer.android.com/reference/android/location/LocationManager.html)),
+and returns the location data to the Flutter app.
+
+* [`android/app/src/main/AndroidManifest.xml`](https://github.com/flutter/flutter/blob/master/examples/hello_services/android/app/src/main/AndroidManifest.xml):
+The manifest file lists the Android activities that perform the services calls and any permissions
+that your app needs to access the services.
+
+* `android/local.properties`: Contains the configuration settings to build your Android app. See
+the `hello_services` [README](https://github.com/flutter/flutter/blob/master/examples/hello_services/README.md)
+to learn more about the properties you can configure.
+
+* [`ios/HelloServices/`](https://github.com/flutter/flutter/tree/master/examples/hello_services/ios/HelloServices):
+This folder contains the Objective-C source files for the iOS-specific part of your app. The
+[`AppDelegate.m`](https://github.com/flutter/flutter/blob/master/examples/hello_services/ios/HelloServices/AppDelegate.m)
+file handles message passing between iOS and the Flutter app, while the [`LocationProvider.m`](https://github.com/flutter/flutter/blob/master/examples/hello_services/ios/HelloServices/LocationProvider.m)
+file implements the code that queries the iOS system location API
+([`CLLocationManager`](https://developer.apple.com/library/ios/documentation/CoreLocation/Reference/CLLocationManager_Class/)).
+
+* `ios/Flutter/Generated.xcconfig`: Contains the configuration settings to build your iOS app. See
+the `hello_services` [README](https://github.com/flutter/flutter/blob/master/examples/hello_services/README.md)
+to learn more about the properties you can configure.
+
+* [`lib/main.dart`](https://github.com/flutter/flutter/blob/master/examples/hello_services/lib/main.dart):
+The main entry point for your Flutter app. This source file is written in Dart syntax. The main
+function uses the [`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html)class,
+which provides a high-level abstraction for passing messages to “hosts” of platform-specific
+services.
+
+## Add a service to your app {#add-a-service-to-your-app}
+
+In your Dart code (`lib/main.dart` file):
+
+1. Define a method to send a request to your target service (e.g. `_getLocation()`). You must encode
+the request as a JSON message. You’ll typically invoke this method in a
+[`State`](http://docs.flutter.io/flutter/material/State-class.html) object, when the user performs
+an operation on the UI.
+2. Define a messageHandler to process the messages returned by the target service
+(e.g. `handleGetRandom()`), then add this handler to the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) object.
```

Comment:
This step is for a completely different thing: exposing services from the cross-platform part of your app to the platform-specific part of your app.  I'd recommend removing this from this document.  It's just confusing the issue.  We might want to remove it from the example as well.